eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/ContentTypeIdentifier.php

```diff
@@ -13,13 +13,38 @@
 use eZ\Publish\Core\Search\Legacy\Content\Common\Gateway\CriteriaConverter;
 use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
 use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use eZ\Publish\SPI\Persistence\Content\Type\Handler as ContentTypeHandler;
 
 /**
  * Content type criterion handler
  */
-class ContentTypeIdentifier extends FieldBase
+class ContentTypeIdentifier extends CriterionHandler
```

Comment:
Yes, it just made sense as we don't need Language Handler or field filters there. I initially added it for common dependency, but `ContentTypeIdentifier` is really not a type of `Field` criterion like others, so should not have been made to inherit from `FieldBase` in the first place.